File: boj/b_1406.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

ls, rs = list(input())[:-1], []
n = int(input())
for _ in range(n):
    inp = list(input().split())
    op = inp[0]
    if op == "L":
        ls and rs.append(ls.pop())
    elif op == "D":
        rs and ls.append(rs.pop())
    elif op == "B":
        ls and ls.pop()
    elif op == "P":
        ls.append(inp[1])
print("".join(ls) + "".join(reversed(rs)))

File: boj/b_1406.js
function solution(init, operations) {
  const [ls, rs] = [init.split(""), []];
  operations.forEach((oper) => {
    oper = oper.split(" ");
    let op = oper[0];
    if (op === "L" && ls.length) rs.push(ls.pop());
    else if (op === "D" && rs.length) ls.push(rs.pop());
    else if (op === "B" && ls.length) ls.pop();
    else if (op === "P") ls.push(oper[1]);
  });
  return ls.join("") + rs.reverse().join("");
}
console.log(solution("abcd", ["P x", "L", "P y"]));
